src/audio/host-zephyr.c

```diff
@@ -459,8 +464,10 @@ static int host_copy_normal(struct host_data *hd, struct comp_dev *dev)
 	hd->partial_size += copy_bytes;
 	buffer_c = buffer_acquire(hd->dma_buffer);
 
-	/* Reload DMA when half of the buffer has been used */
-	if (hd->partial_size >= buffer_c->stream.size >> 1) {
+	/* Reload DMA when the threshold is reached, add a 2ms safety margin */
+	if (!IS_ENABLED(CONFIG_HOST_DMA_RELOAD_DELAY_ENABLE) ||
+	    buffer_c->stream.size < hd->period_bytes << 3 ||
+	    buffer_c->stream.size - hd->partial_size <= (2 + threshold) * hd->period_bytes) {
```

Comment:
Some comment on "period_bytes << 3" would be nice as well.

Reply:
@kv2019i comment updated